src/forcing/ForcingsEngineLumpedDataProvider.cpp

```diff
@@ -42,9 +42,19 @@ Provider::ForcingsEngineLumpedDataProvider(
         id_dim
     );
 
+    // temporary map to ensure uniqueness
+    std::unordered_map<int, int> uniq;
+    uniq.reserve(id_dim);
     var_divides_.reserve(id_dim);
     for (int i = 0; i < id_dim; ++i) {
-        var_divides_[cat_id[i]] = i;
+        const auto id = cat_id[i];
+    
+        uniq[id]++;
+        if (uniq[id] > 1) {
+            throw std::runtime_error{"Non-unique catchment ID found in lumped forcings engine domain: cat-" + std::to_string(id)};
```

Comment:
Let's not encode an assumption that the prefix is `cat-` here. Preferable to just print the number exactly as is.

Otherwise, this looks good.

Reply:
Revised in [8e8fa99](https://github.com/NOAA-OWP/ngen/pull/720/commits/8e8fa9926e97f38f51c9f740be6c43651369e23b)